hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/parser/HoodieDateTimeParserImpl.java

```diff
@@ -51,22 +47,21 @@ private DateTimeFormatter getInputDateFormatter() {
       throw new IllegalArgumentException(Config.TIMESTAMP_INPUT_DATE_FORMAT_PROP + " configuration is required");
     }
 
-    DateTimeFormatter formatter = new DateTimeFormatterBuilder()
-        .append(
-        null,
-        Arrays.stream(
-          this.configInputDateFormatList.split(super.configInputDateFormatDelimiter))
-          .map(String::trim)
-          .map(DateTimeFormat::forPattern)
-          .map(DateTimeFormatter::getParser)
-          .toArray(DateTimeParser[]::new))
+    DateTimeFormatterBuilder formatterBuilder = new DateTimeFormatterBuilder();
+    Arrays.stream(this.configInputDateFormatList.split(super.configInputDateFormatDelimiter))
+        .map(String::trim)
+        .map(DateTimeFormatter::ofPattern)
+        .forEach(formatterBuilder::appendOptional);
+    DateTimeFormatter formatter = formatterBuilder
+        .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+        .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+        .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
         .toFormatter();
     if (this.inputDateTimeZone != null) {
       formatter = formatter.withZone(this.inputDateTimeZone);
     } else {
-      formatter = formatter.withOffsetParsed();
+      formatter = formatter.withZone(ZoneId.systemDefault());
```

Comment:
this is not fully compatible change: joda has a feature to infer timezone from input string but java time does not. so it supplies system default timezone.